pkg/acceptance/acceptance.go

```diff
@@ -84,6 +84,34 @@ func RunAcceptance(ctx context.Context, logger kitlog.Logger) {
 					Expect(txact.Rollback()).To(Succeed(), "transaction to rollback, as connection should have been uninterrupted")
 				})
 			})
+
+			Context("With unhealthy keeper", func() {
+				var (
+					asyncKeeper string
+				)
+
+				BeforeEach(func() {
+					asyncKeeper = mustClusterdata(ctx, client).AsynchronousStandbys()[0].Spec.KeeperUID
+					Expect(
+						execCommand(ctx, "docker-compose", "exec", asyncKeeper, "pkill", "-STOP", "stolon-keeper"),
+					).To(Succeed(), "stopping %s stolon-keeper service", asyncKeeper)
+
+					Eventually(func() error { return mustClusterdata(ctx, client).CheckHealthy() }).Should(
+						Not(Succeed()), "timed out waiting for > min keepers to become unhealthy",
+					)
+				})
+
+				AfterEach(func() {
+					Expect(
+						execCommand(ctx, "docker-compose", "exec", asyncKeeper, "pkill", "-CONT", "stolon-keeper"),
+					).To(Succeed(), "starting %s stolon-keeper service", asyncKeeper)
+				})
+
+				It("Fails to failover without minimum number of healthy keepers", func() {
+					Expect(err).To(HaveOccurred(), "failover to have failed due to too many unhealthy keepers")
```

Comment:
This annotation is a bit out of place. If you think about this appearing with your failure, and how it can help you, phrasing it like this might be better:

```
failover succeeded when it should have failed due to unhealthy keeper
```

wdyt?

Reply:
Sounds better to me!